### PR TITLE
Refactor datetime formatting to use Django's localtime

### PR DESCRIPTION
**Pull request by willbarton**
TLDR: This PR replaces adds Django's `localtime` template tag/filter to our Jinja2 environment and then refactors the `time.render()` and `datetime.format_time()` macros to use `localtime` for timezone handling and changes everywhere that `time.render()` was called with a separate timezone. This should maintain the existing expectations of rendering.

Longer:

A side effect [our recent effort to remove the custom sharing code](https://github.com/cfpb/cfgov-refresh/pull/2827/commits/26e53bdd4ad786436e4457d54243323099263d85#diff-8e8a3e28d46ee12516bea4f5b50bf5a4L220) is that event starting times were showing up in UTC:

image

This is because the `get_appropriate_page_version()` method that was removed hide a quirk of timezone handling: By always getting the latest page revision from JSON and then converting it to a Wagtail Page, it was converting the UTC datetime to the configured `settings.TIME_ZONE`. When that was removed, we're left with UTC. 

Additionally, our `date` filter in Jinja2 templates was not Django's `date` filter, but our own [custom date filter from sheerlike](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/sheerlike/templates.py#L19). The sheerlike `get_date_string()` doesn't do any special timezone handling, and doesn't check `settings.TIME_ZONE` but relies on a timezone being passed in.

With this change we have the desired outcome on the event:

image

## Changes

- Replace sheerlike's `date` filter with Django's `date` filter.

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

File: cfgov/v1/__init__.py [hunk 1]
```diff
@@ -58,10 +63,12 @@ def environment(**options):
         'get_protected_url': get_protected_url,
         'get_latest_activities': get_latest_activities,
         'get_snippets': get_snippets,
+        'localtime': template_localtime,
     })
 
     env.filters.update({
         'linebreaksbr': linebreaksbr,
+        'localtime': template_localtime,
```
Comment (richaagarwal) [inline cfgov/v1/__init__.py]: can we just call `template_localtime` directly so we don't have to do an extra look up here?

Comment (richaagarwal) [inline cfgov/v1/__init__.py]: sorry assuming we have to register it, can we call it the same thing, e.g. `template_localtime: template_localtime`

Comment (willbarton) [inline cfgov/v1/__init__.py]: What do you mean?

Comment (willbarton) [inline cfgov/v1/__init__.py]: [Django includes it in Django templates as `localtime`, not `template_localtime`](https://docs.djangoproject.com/en/1.10/topics/i18n/timezones/#localtime). Here I'm including it in the Jinja2 environment with the same name Django uses in its templates.

Comment (richaagarwal) [inline cfgov/v1/__init__.py]: Ah okay, interesting!  Sounds good then.
